cap/modules/mail/attributes.py

```diff
@@ -0,0 +1,285 @@
+    """
+    For each condition we have a group of checks to perform, which should
+    return True/False
+
+    Each condition has the following:
+    - `checks`: the checks to be evaluated
+    - `op`: the operation to be used for all the checks (and/or)
+    - `mails`: the mail list
+
+    Each check has their own attributes:
+    - `if`: the type of check (e.g. exists, is_in, etc)
+    - `value`: the result
+    - `path`: the path that gets the field to be evaluated
+
+    An example of conditions:
+    {
+      "type": "condition",
+      "checks": [{
+        "path": "parton_distribution_functions",
+        "if": "exists",
+        "value": true
+      }],
+      "mails": {
+        "default": ["pdf-forum-placeholder@cern.ch"]
+      }
+    }
+
+    Nested checks are also allowed.
+    """
+    # for each condition we have a group of checks to perform
+    # all of the checks should give a True/False result
+    operator = condition.get('op', 'and')
+    checks = condition['checks']
+    check_results = []
+
+    for check in checks:
+        # get the method to apply, and use it on the required path/value
+        if check.get('if'):
+            method = CONDITION_METHODS[check['if']]
+            check_results.append(
+                method(record, check.get('path'), check.get('value', True))
+            )
+        else:
+            check_results.append(
+                check_condition(record, check)
+            )
+
+    # we check the validity of the condition depending on the operator:
+    # - if 'and', then we need all the items to be true
+    # - if 'or' we need at least 1 item to be true
+    if (operator == 'and' and all(check_results)) or \
+            (operator == 'or' and any(check_results)):
+        return True
+    return False
+
+
+def get_recipients_from_config(record, config):
+    """
+    The `recipients` field differentiates 3 categories:
+    - recipients
+    - cc
+    - bcc
+    All 3 can be used to send a mail, and have their own mails and rules about
+    how they will be added.
+
+    The rules are in 3 categories:
+    - `default`: the mails in the list will be added
+    - `method`: a method that returns a list of mail (for complicated options)
+    - `conditions`: mails will be added if a certain condition is true
+
+    An example config of recipients:
+    "recipients": {
+      "bcc": [
+        {
+          "type": "method",
+          "method": "get_owner"
+        }, {
+          "type": "default",
+          "mails": {
+            "default": ["some-recipient-placeholder@cern.ch"],
+            "formatted": [{
+              "template": "{% if cadi_id %}hn-cms-{{ cadi_id }}@cern.ch{% endif %}",  # noqa
+              "ctx": [{
+                "name": "cadi_id",
+                "type": "path",
+                "path": "analysis_context.cadi_id"
+              }]
+            }]
+          }
+        }, {
+          "type": "condition",
+          ...
+        }
+      ]
+    }
+    """
```

Comment:
# Accept strings
        # If typeof string, check regex and add to list

        # If dict continue

        # if key 'checks', do check_condition

        # If key 'method' return method

        # then if 'emails'